google/cloud/testing_util/validate_metadata.cc

```diff
@@ -315,14 +314,30 @@ void ValidateMetadataFixture::IsContextMDValid(
   auto actual = dist == 0 ? RoutingHeaders{}
                           : ExtractMDFromHeader(param_header.first->second);
 
-  // Extract expectations on `x-goog-request-params` from the `google.api.http`
-  // annotation on the specified method.
+  auto const* method =
+      DescriptorPool::generated_pool()->FindMethodByName(method_name);
+  ASSERT_THAT(method, NotNull()) << "Method " + method_name + " is unknown.";
+
+  // Extract expectations on `x-goog-request-params` from the
+  // `google.api.routing` or `google.api.http` annotation on the specified
+  // method.
   auto expected = ExtractRoutingHeaders(method, request, resource_name);
 
+  // Do not verify routing parameters for streaming writes, because it is not
+  // well defined.
+  if (method->client_streaming()) {
+    GCP_LOG(INFO) << "`x-goog-request-params` header not verified for "
+                  << method_name << ", because it is a streaming write.";
+    return;
+  }
+
   // Check if the metadata in the context satisfied the expectations.
+  std::vector<Matcher<std::pair<std::string, std::string>>> matchers;
+  matchers.reserve(expected.size());
   for (auto const& param : expected) {
-    EXPECT_THAT(actual, Contains(Pair(param.first, MatchesGlob(param.second))));
+    matchers.push_back(Pair(param.first, MatchesGlob(param.second)));
   }
+  EXPECT_THAT(actual, UnorderedElementsAreArray(matchers));
```

Comment:
This seems like something that should be expressible directly in gmock, not that I know how to do so off the top of my head.

Reply:
Hmm. It may be `UnorderedPointwise(...)`. Let me see if it works.

Reply:
Did you have any luck?  I don't want to hold you up unnecessarily.

Reply:
After chatting, LGTM.

Reply:
To summarize, this approach with `UnorderedPointwise` works: https://github.com/dbolduc/google-cloud-cpp/commit/bbc062015c88e0a32f682acc076d4c00e06b7bc6

I think the code is cleaner. But I think the error messages are less helpful.

I could not find a good way to generate error messages using `UnorderedPointwise` (trying things like `ExplainMatchResult` outputting to `*result_listener`).

I will leave the PR as is because there are other things I could be doing instead of trying to achieve "gMock correctness".